extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapLoaderTest.java

```diff
@@ -0,0 +1,462 @@
+    public void givenRowDoesNotExist_whenLoad_thenReturnNull() throws Exception {
+        createTable(mapName);
+        mapLoader = createMapLoader();
+
+        GenericRecord record = mapLoader.load(0);
+        assertThat(record).isNull();
+    }
+
+    @Test
+    public void givenRow_whenLoadAll_thenReturnMapWithGenericRecord() throws Exception {
+        createTable(mapName);
+        mapLoader = createMapLoader();
+
+        insertItems(mapName, 1);
+
+        Map<Integer, GenericRecord> records = mapLoader.loadAll(newArrayList(0));
+
+        assertThat(records).containsKey(0);
+
+        GenericRecord record = records.values().iterator().next();
+        records.values().iterator().next();
+        assertThat(record.getInt32("id")).isZero();
+        assertThat(record.getString("name")).isEqualTo("name-0");
+    }
+
+    @Test
+    public void givenRowAndIdColumn_whenLoadAll_thenReturnGenericRecord() throws Exception {
+        createTable(mapName, "\"person-id\" INT PRIMARY KEY", "name VARCHAR(100)");
+        insertItems(mapName, 1);
+
+        Properties properties = new Properties();
+        properties.setProperty(DATA_LINK_REF_PROPERTY, TEST_DATABASE_REF);
+        properties.setProperty(ID_COLUMN_PROPERTY, "person-id");
+        mapLoader = createMapLoader(properties, hz);
+        GenericRecord record = mapLoader.loadAll(newArrayList(0)).get(0);
+
+        assertThat(record.getInt32("person-id")).isZero();
+        assertThat(record.getString("name")).isEqualTo("name-0");
+    }
+
+    @Test
+    public void givenRowDoesNotExist_whenLoadAll_thenReturnEmptyMap() throws Exception {
+        createTable(mapName);
+        mapLoader = createMapLoader();
+
+        Map<Integer, GenericRecord> records = mapLoader.loadAll(newArrayList(0));
+        assertThat(records).isEmpty();
+    }
+
+    @Test
+    public void givenRow_whenLoadAllKeys_thenReturnKeys() throws Exception {
+        createTable(mapName);
+        mapLoader = createMapLoader();
+
+        insertItems(mapName, 1);
+
+        List<Integer> ids = newArrayList(mapLoader.loadAllKeys());
+        assertThat(ids).contains(0);
+    }
+
+    @Test
+    public void givenRowAndIdColumn_whenLoadAllKeys_thenReturnKeys() throws Exception {
+        createTable(mapName, "\"person-id\" INT PRIMARY KEY", "name VARCHAR(100)");
+        insertItems(mapName, 1);
+
+        Properties properties = new Properties();
+        properties.setProperty(DATA_LINK_REF_PROPERTY, TEST_DATABASE_REF);
+
+        properties.setProperty(ID_COLUMN_PROPERTY, "person-id");
+        mapLoader = createMapLoader(properties, hz);
+
+        List<Integer> ids = newArrayList(mapLoader.loadAllKeys());
+        assertThat(ids).contains(0);
+    }
+
+    @Test
+    public void givenNoRows_whenLoadAllKeys_thenEmptyIterable() throws Exception {
+        createTable(mapName);
+        mapLoader = createMapLoader();
+
+        Iterable<Integer> ids = mapLoader.loadAllKeys();
+        assertThat(ids).isEmpty();
+    }
```

Comment:
All these tests are duplicated in `GenericMapStoreTest` and `GenericMapLoaderTest`.

We should make the GenericMapStoreTest extend the `GenericMapLoaderTest` to remove the duplication.

Reply:
OK. GenericMapStoreTest  extends GenericMapLoaderTest

Reply:
The tests are still duplicated.

Reply:
Removed all duplicate tests showed by IntelliJ from GenericMapStoreTest